distributed/scheduler.py

```diff
@@ -2126,6 +2126,18 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
             n_workers: Py_ssize_t = len(worker_pool_dv)
             if n_workers < 20:  # smart but linear in small case
                 ws = min(worker_pool.values(), key=operator.attrgetter("occupancy"))
+                if ws.occupancy == 0:  # special case to use round-robin
+                    wp_vals = worker_pool.values()
+                    start = self._n_tasks % n_workers
+                    for i in range(start, n_workers):
```

Comment:
I recommend having this go from 0..n and then indexing with `start + i % n`.  This should hopefully allow you to collapse the two for loops into one, and ensure greater consistency between the bodies of those loops in the future.

Reply:
Is the `n` you're using referring to `self._n_tasks`?